collector/src/lib.rs

```diff
@@ -496,17 +502,27 @@ impl<V: vdaf::Collector> Collector<V> {
             .headers()
             .get(CONTENT_TYPE)
             .ok_or(Error::BadContentType(None))?;
-        if content_type != CollectionMessage::<TimeInterval>::MEDIA_TYPE {
+        if content_type != CollectResp::<TimeInterval>::MEDIA_TYPE {
             return Err(Error::BadContentType(Some(content_type.clone())));
         }
 
-        let collect_response = CollectionMessage::<Q>::get_decoded(&response.bytes().await?)?;
+        let collect_response = CollectResp::<Q>::get_decoded(&response.bytes().await?)?;
         if collect_response.encrypted_aggregate_shares().len() != 2 {
             return Err(Error::AggregateShareCount(
                 collect_response.encrypted_aggregate_shares().len(),
             ));
         }
 
+        let mut aad = Vec::new();
+        aad.extend(self.parameters.task_id.as_ref());
+        aad.extend(
+            &BatchSelector::<Q>::new(Q::batch_identifier_for_collection(
+                &job.query,
+                &collect_response,
+            ))
+            .get_encoded(),
+        );
```

Comment:
Although I suppose there's multiple ways of deriving the AAD that make having a single common function impossible. Still, could have consider having a suite of AAD derivation functions for each use case.

I don't feel strongly about this, can take it or leave it.

Reply:
Yeah, this is a good idea. I introduced new free functions rather than `InputShareAad`/`AggregateShareAad` types since the encoding doesn't quite fit with what's expected by `Encode`/`Decode` -- `Decode` in particular expects to be able to read a message from a stream, which isn't possible with the DAP-02 encoding of the AAD values. (This is only a semantic issue of course, but I suppose it makes sense to me.)